benches/rustposix-criterion.rs

```diff
@@ -1,68 +1,157 @@
+        {
+            unsafe{
+                libc::getpid();
+                libc::getppid();
+                libc::getgid();
+                libc::getegid();
+                libc::getuid();
+                libc::geteuid();
+            }
+        }
+    ));
+    group.finish();
+
+
+
+
+    // --- COMPARING open / close CALLS ACROSS Lind + Native OS kernel ---
+    let mut group = c.benchmark_group("Compare fs:open+close");
+
+    // Should be similar.  Use a linear scale...
+    group.plot_config(criterion::PlotConfiguration::default().summary_scale(criterion::AxisScale::Linear));
+
+    // Let's see how fast various file system calls are
+    group.bench_function("Lind open+close", |b| b.iter(|| 
+        {
+            let fd = cage.open_syscall("foo",O_CREAT | O_TRUNC | O_WRONLY,S_IRWXA);
+            cage.close_syscall(fd);
+        }
+    ));
+
+    // For comparison let's time the native OS...
+    group.bench_function("Native OS kernel open+close", |b| b.iter(|| 
         {
-            a.getpid_syscall();
-            a.getppid_syscall();
-            a.getgid_syscall();
-            a.getegid_syscall();
-            a.getuid_syscall();
-            a.geteuid_syscall();
+            unsafe{
+                let fd = libc::open(CString::new("/tmp/foo").unwrap().as_ptr(),O_CREAT | O_TRUNC | O_WRONLY,S_IRWXA);
+                libc::close(fd);
+            }
         }
     ));
+    group.finish();
+
+    
+
+    
+
+    // --- COMPARING read + write + lseek CALLS ACROSS Lind + Native OS kernel ---
+    let mut group = c.benchmark_group("Compare fs:write+read");
+
+    // Should be similar.  Use a linear scale...
+    group.plot_config(criterion::PlotConfiguration::default().summary_scale(criterion::AxisScale::Linear));
+
+    let fd = cage.open_syscall("foo",O_CREAT | O_TRUNC | O_WRONLY,S_IRWXA);
+    // Let's see how fast various file system calls are
+    group.bench_function("Lind write+read", |b| b.iter(|| 
+        {
+            let _ = cage.write_syscall(fd,CString::new("Well, hello there!!!").unwrap().as_ptr(),20);
+            cage.lseek_syscall(fd,0,SEEK_SET);
+            let mut read_buffer = sizecbuf(20);
+            cage.read_syscall(fd,read_buffer.as_mut_ptr(), 20);
+            cage.lseek_syscall(fd,0,SEEK_SET);
+        }
+    ));
+
+    cage.close_syscall(fd);
+    cage.unlink_syscall("foo");
+
+
+    let fd: c_int;
+
+    unsafe {
+        fd = libc::open(CString::new("/tmp/foo").unwrap().as_ptr(),O_CREAT | O_TRUNC | O_WRONLY,S_IRWXA);
+    }
+
+    // For comparison let's time the native OS...
+    group.bench_function("Native OS kernel read+write", |b| b.iter(|| 
+        {
+            unsafe{
+                let _ = libc::write(fd,CString::new("Well, hello there!!!").unwrap().as_ptr() as *const c_void,20);
+                let mut read_buffer = sizecbuf(20);
+                libc::lseek(fd,0,SEEK_SET);
+                cage.read_syscall(fd,read_buffer.as_mut_ptr(), 20);
```

Comment:
If we want to test Native OS, should we use `libc::read(fd, read_buffer.as_mut_ptr(), 20) `here?

Reply:
```suggestion
                libc::read(fd,read_buffer.as_mut_ptr() as *mut c_void, 20);
```

Reply:
Thanks, fixed!